src/lib/Repository/UiRelationService.php

```diff
@@ -62,19 +61,16 @@ public function loadRelations(VersionInfo $versionInfo)
     /**
      * Loads reverse relations for a piece of content.
      * Retrieves reverse relations and then sets the field definition and the content type name.
-     * Returns false if the user doesn't have permission to load reverse relations.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the user is not allowed to read reverse relations.
      *
      * @param ContentInfo $contentInfo
      *
-     * @return UiRelation[]|bool
+     * @return UiRelation[]
      */
     public function loadReverseRelations(ContentInfo $contentInfo)
     {
-        try {
```

Comment:
Hmm, I may not have explained correctly. It is good that you're using the resolver, but I was suggesting that we check for permissions _inside this method_ instead of in the action.

As said in another comment, if we consider that those services implement the business logic of the hybrid UI, it makes sense.